answers/views.py

```diff
@@ -41,8 +42,14 @@ def post(self, request, hunk_pk, answer_pk):
         status_form = UpdateAnswerStatusForm(request.POST)
 
         if status_form.is_valid():
-            answer = get_object_or_404(Answer, pk=answer_pk)
-            answer.set_status(status_form.cleaned_data["status"])
+            guess = get_object_or_404(Answer, pk=answer_pk)
+            status = status_form.cleaned_data["status"]
+            if status == Answer.CORRECT and len(guess.puzzle.answer) > 0:
+                messages.warning(request,
+                    '{} was already marked as solved with the answer "{}"\n'.format(guess.puzzle, guess.puzzle.answer) +
+                    'We won\'t stop ya, but please think twice.')
```

Comment:
I think this message gets shown after the form has already been submitted. Is it possible to show this warning (perhaps a pop-up confirmation box) when Submit is pressed and before the POST request is sent? The confirmation box could say something like what you have above plus "Are you sure you still want to submit this answer?"

Reply:
I'm not really familiar with the javascript, could we go with this for now and add it later?

Reply:
Yeah, sure, this is just a nice-to-have, not critical. Fine with this for now.